osu.Game/Overlays/MusicController.cs

```diff
@@ -256,8 +257,10 @@ private PreviousTrackResult prev(bool allowProtectedTracks)
                 playableSet = getNextRandom(-1, allowProtectedTracks);
             else
             {
-                playableSet = getBeatmapSets().TakeWhile(i => !i.Value.Equals(current?.BeatmapSetInfo)).LastOrDefault(s => !s.Value.Protected || allowProtectedTracks)
-                              ?? getBeatmapSets().LastOrDefault(s => !s.Value.Protected || allowProtectedTracks);
+                var beatmapSets = getBeatmapSets(allowProtectedTracks);
+
+                playableSet = beatmapSets.TakeWhile(i => !i.Value.Equals(current?.BeatmapSetInfo)).LastOrDefault(s => !s.Value.Protected || allowProtectedTracks)
+                              ?? beatmapSets.LastOrDefault(s => !s.Value.Protected || allowProtectedTracks);
```

Comment:
Do we still need the lambdas here if they've been moved into `getBeatmapSets()`?

Reply:
Ah, no, didn't see these ones.